From 68954e3eddf2ff7aebcfadf7109bb52b48b19024 Mon Sep 17 00:00:00 2001
From: Finn Haedicke <finn@informatik.uni-bremen.de>
Date: Thu, 17 Jun 2010 11:07:53 +0200
Subject: [PATCH 01/14] moved typedefs Btor and BtorExp to seperate header

This way only a single small header has to be included from boolector.h
---
 boolector.h |    3 +--
 btordefs.h  |    7 +++++++
 btorexp.h   |    5 +----
 3 files changed, 9 insertions(+), 6 deletions(-)
 create mode 100644 btordefs.h

diff --git a/boolector.h b/boolector.h
index 9f7c362..5f80aae 100644
--- a/boolector.h
+++ b/boolector.h
@@ -26,8 +26,7 @@
 Boolector is copyrighted 2007 - 2009 by Robert Brummayer, Armin
 */
 
-#include "btorexp.h"
-
+#include "btordefs.h"
 #include <stdio.h>
 
 /*------------------------------------------------------------------------*/
diff --git a/btordefs.h b/btordefs.h
new file mode 100644
index 0000000..f02a0b9
--- /dev/null
+++ b/btordefs.h
@@ -0,0 +1,7 @@
+#ifndef BTORDEFS_H_INCLUDED
+#define BTORDEFS_H_INCLUDED
+
+typedef struct BtorExp BtorExp;
+typedef struct Btor Btor;
+
+#endif
diff --git a/btorexp.h b/btorexp.h
index 0600586..5611c64 100644
--- a/btorexp.h
+++ b/btorexp.h
@@ -20,6 +20,7 @@
 #ifndef BTOREXP_H_INCLUDED
 #define BTOREXP_H_INCLUDED
 
+#include "btordefs.h"
 #include "btorstack.h"
 #include "btormem.h"
 #include "btoraigvec.h"
@@ -40,10 +41,6 @@
 /* Declarations                                                           */
 /*------------------------------------------------------------------------*/
 
-typedef struct BtorExp BtorExp;
-
-typedef struct Btor Btor;
-
 BTOR_DECLARE_STACK (ExpPtr, BtorExp *);
 
 BTOR_DECLARE_QUEUE (ExpPtr, BtorExp *);
-- 
1.7.3.4

